lib/sna/auth/email.ex

```diff
@@ -0,0 +1,40 @@
+defmodule Sna.Auth.Mail do
+
+  @moduledoc """
+
+  Sna.Auth.Mail will store email and token:
+
+  1. when an user comes and want to use email authentication, an email
+     is firstly checked.
+
+  2. if the mail is valid, a token is generated and the email is
+     stored temporarily in the ETS cache. A timer process is started
+     for some minutes.
+
+  3. if the token is valid (checked by the user), the mail is stored
+     in the underlaying database else, after some minutes, the mail
+     and token are rejected.
+
+  4. after validation, Sna.Auth.Mail process will return the session
+     identifier.
+
+  """
+  
+  use GenServer
+
+  def start_link(), do: start_link([])
+  def start_link(args), do: GenServer.start_link(__MODULE__, args, name: __MODULE__)
+
+  def init(_args), do: {:ok, :ets.new(__MODULE__, [:private])}
```

Comment:
I create a new ETS table (that a local cache mechanism), `:ets.new/2` will return a reference to the table. So, our process will have only a reference to this table as state.